app/assets/javascripts/govuk_publishing_components/components/feedback.js

```diff
@@ -49,6 +50,22 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         revealInitialPrompt()
       })
 
+      this.$pageIsNotUsefulButton.on('click', function (e) {
+        var cookie = document.cookie.split("; ") ;
+        var i ;
+        var clientIDFound = false;
+        for (i = 0; i < cookie.length; i++) {
+          if (cookie[i].startsWith("_ga")) {
+            document.gaClientId = cookie[i].split("=")[1].split(".").slice(-2).join(".")
```

Comment:
You're trying to set a global variable? The traditional way of doing this is to use `window`, i.e. `window.gaClientId = ...`.

But it's generally better to use scoped attributes (`this`). On line  24 you can have `this.gaClientId = undefined;`, which means on this line (59) you can do `this.gaClientId = ...`.

But then you're going to face the [`this` problem](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this), because your `on('click'` declaration is taking an "anonymous function" as a callback (the `function (e) {}` bit). So the value of `this` is pointing to the anonymous function and not the `Modules.Feedback` object.

So you'll want to use `that.gaClientId` (we set `that = this` on line 25 to get around this issue, it's quite a common workaround). The alternative is [binding your function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind).